src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -412,6 +412,14 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		}
 
 		let newProfile = new ConnectionProfile(this._capabilitiesService, model || providerName);
+		if (!model.password && defaultAuthenticationType === AuthenticationType.SqlLogin && authenticationTypeName === AuthenticationType.SqlLogin) {
+			try {
+				await this._connectionManagementService.addSavedPassword(newProfile);
+			}
+			catch (err) {
+				onUnexpectedError(err);
```

Comment:
Would be good to create another error wrapping this to clearly indicate what failed (something like `Error filling in password for connection dialog model`)